modules/data/event-hub/main.bicep

```diff
@@ -0,0 +1,245 @@
+  captureDescriptionEncoding: contains(eventHub.value, 'captureDescriptionEncoding') ? eventHub.value.captureDescriptionEncoding : 'Avro'
+  captureDescriptionIntervalInSeconds: contains(eventHub.value, 'captureDescriptionIntervalInSeconds') ? eventHub.value.captureDescriptionIntervalInSeconds : 300
+  captureDescriptionSizeLimitInBytes: contains(eventHub.value, 'captureDescriptionSizeLimitInBytes') ? eventHub.value.captureDescriptionSizeLimitInBytes : 314572800
+  captureDescriptionSkipEmptyArchives: contains(eventHub.value, 'captureDescriptionSkipEmptyArchives') ? eventHub.value.captureDescriptionSkipEmptyArchives : false
+}]
+
+var varEventHubAuthorizationRules =  [for eventHubAuthorizationRule in items(eventHubAuthorizationRules): {
+  eventHubAuthorizationRuleName: eventHubAuthorizationRule.key
+  rights: contains(eventHubAuthorizationRule.value, 'rights') ? eventHubAuthorizationRule.value.rights : []
+  eventHubNamespaceName: eventHubAuthorizationRule.value.eventHubNamespaceName
+  eventHubName: eventHubAuthorizationRule.value.eventHubName
+}]
+
+var varConsumerGroups =  [for consumerGroup in items(consumerGroups): {
+  consumerGroupName: consumerGroup.key
+  eventHubNamespaceName: consumerGroup.value.eventHubNamespaceName
+  eventHubName: consumerGroup.value.eventHubName
+  userMetadata: contains(consumerGroup.value, 'userMetadata') ? consumerGroup.value.userMetadata : ''
+}]
+
+var varDiagnosticSettings = [for diagnosticSetting in items(diagnosticSettings): {
+  diagnosticSettingName: diagnosticSetting.key
+  diagnosticEnableNamespaceName: diagnosticSetting.value.diagnosticEnablenamespaceName
+  diagnosticStorageAccountId: contains(diagnosticSetting.value, 'diagnosticStorageAccountId') ? diagnosticSetting.value.diagnosticStorageAccountId : ''
+  diagnosticWorkspaceId: contains(diagnosticSetting.value, 'diagnosticWorkspaceId') ? diagnosticSetting.value.diagnosticWorkspaceId : ''
+  diagnosticEventHubAuthorizationRuleId: contains(diagnosticSetting.value, 'diagnosticEventHubAuthorizationRuleId') ? diagnosticSetting.value.diagnosticEventHubAuthorizationRuleId : ''
+  diagnosticEventHubName: contains(diagnosticSetting.value, 'diagnosticEventHubName') ? diagnosticSetting.value.diagnosticEventHubName : ''
+  diagnosticsMetrics: contains(diagnosticSetting.value, 'diagnosticsMetrics') ? diagnosticSetting.value.diagnosticsMetrics : []
+  diagnosticsLogs: contains(diagnosticSetting.value, 'diagnosticsLogs') ? diagnosticSetting.value.diagnosticsLogs : []
+}]
+
+resource cluster 'Microsoft.EventHub/clusters@2021-11-01' = if (clusterName != 'null' )  {
+  name: clusterName
+  location: location
+  tags: tags
+  sku: {
+    name: 'Dedicated'
+    capacity: clusterCapacity
+  }
+  properties: {}
+  dependsOn: [
+    resourceGroup()
+  ]
+}
+
+resource eventHubNamespace 'Microsoft.EventHub/namespaces@2021-11-01' = [for varEventHubNamespace in varEventHubNamespaces: if (!empty(eventHubNamespaces)) {
```

Comment:
```suggestion
resource eventHubNamespace 'Microsoft.EventHub/namespaces@2021-11-01' = [for varEventHubNamespace in varEventHubNamespaces: {
```

Reply:
don't need to check if group is empty, this is automatically handled by the for-in loop.

Reply:
done